### PR TITLE
Adding Supplier List & List Type managers

### PR DESCRIPTION
**Pull request by Yvler**

### CODE REVIEW

File: lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+
+
+	/**
+	 * Removes old entries from the storage.
+	 *
+	 * @param integer[] $siteids List of IDs for sites whose entries should be deleted
+	 */
+	public function cleanup( array $siteids )
+	{
+		$path = 'classes/product/manager/list/type/submanagers';
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php]: Must be "classes/supplier/...". This was a bug in the customer list class which have been already fixed :-)

File: lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+		foreach( $this->_getContext()->getConfig()->get( $path, array() ) as $domain ) {
+			$this->getSubManager( $domain )->cleanup( $siteids );
+		}
+
+		$this->_cleanup( $siteids, 'mshop/product/manager/list/type/default/item/delete' );
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php]: Must be "mshop/supplier..." too

File: lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+		 * The search keys from sub-managers can be normally used in the
+		 * manager as well. It allows you to search for items of the manager
+		 * using the search keys of the sub-managers to further limit the
+		 * retrieved list of items.
+		 *
+		 * @param array List of sub-manager names
+		 * @since 2014.03
+		 * @category Developer
+		 */
+		$path = 'classes/product/manager/list/type/submanagers';
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php]: Must be "classes/supplier/..." too

File: lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+
+
+	/**
+	 * Returns a new manager for Supplier list type extensions.
+	 *
+	 * @param string $manager Name of the sub manager type in lower case
+	 * @param string|null $name Name of the implementation, will be from configuration (or Default) if null
+	 * @return MShop_Common_Manager_Interface Manager for different extensions, e.g types, lists etc.
+	 */
+	public function getSubManager( $manager, $name = null )
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php]: Please replace all occurrences of "product" with "supplier" in this method

File: lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+		return $this->_getSubManager( 'product', 'list/type/' . $manager, $name );
+	}
+
+
+	/**
+	 * Returns the config path for retrieving the configuration values.
+	 *
+	 * @return string Configuration path
+	 */
+	protected function _getConfigPath()
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Supplier/Manager/List/Type/Default.php]: Same in this method (product vs. supplier)

**Comment by aimeos** [comment]
Please replace all copyright lines in each new file with yours or the Aimeos' ones if you wan't it belong to the Aimeos project :-)

**Comment by Yvler** [comment]
Re-do pull request, changing "product" to "supplier"

**Comment by aimeos** [comment]
If you want the tests to succeed, you should copy and adapt the lib/mshoplib/setup/unittest/data/customer-list.php as well as the lib/mshoplib/setup/unittest/CustomerListAddTestData.php too

**Comment by Yvler** [comment]
New pull request :) files changed. I hope the test files are right this way..

File: lib/mshoplib/setup/unittest/data/supplier-list.php [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+ * @license LGPLv3, http://opensource.org/licenses/LGPL-3.0
+ */
+
+return array (
+    'supplier/list/type' => array (
+        'text/default' => array( 'domain' => 'text', 'code' => 'default', 'label' => 'Default', 'status' => 1 ),
+    ),
+
+    'supplier/list' => array (
+        array( 'parentid' => 'supplier/UTC001', 'typeid' => 'text/default', 'domain' => 'text', 'refid' => 'text/supplier/description', 'start' => '2010-01-01 00:00:00', 'end' => '2022-01-01 00:00:00', 'config' => array(), 'pos' => 0, 'status' => 1 ),
```
Comment (aimeos) [inline lib/mshoplib/setup/unittest/data/supplier-list.php]: This must be 'parentid' => 'supplier/unitCode001' to match the supplier code in the supplier.php file

**Comment by aimeos** [comment]
Almost done :-)
You can run the unit tests locally to fix the remaining errors:
- Create a new database
- Add the name and credentials in "./config/resource.php" (resource.php-dist is a template for this)
- Execute "phing" on the command line in your aimeos-core directory (you need phing for this: http://phing.info)

**Comment by Yvler** [comment]
I'm on my work environment and don't have access to install here. Can you check these test files instead?

**Comment by aimeos** [comment]
I can't push to your PR directly, so I had to create a new branch and had to make the changes there. This branch containing your changes is merged now, so I close this PR

Thanks a lot for your great work! :-)

**Pull request closed without merging**